presto-release/src/main/java/com/facebook/presto/release/tasks/GenerateReleaseNotesTask.java

```diff
@@ -0,0 +1,443 @@
+        List<String> commitIds = Splitter.on("\n")
+                .trimResults()
+                .omitEmptyStrings()
+                .splitToList(gitCommands.log(
+                        format(
+                                "%s/release-%s..%s/release-%s",
+                                repository.getUpstreamName(),
+                                version.getLastVersion().getVersion(),
+                                repository.getUpstreamName(),
+                                version.getVersion()),
+                        "--format=%H",
+                        "--date-order"));
+
+        log.info("Fetching Github commits");
+        List<Commit> commits = githubAction.listCommits(commitIds.get(0), commitIds.get(commitIds.size() - 1));
+
+        log.info("Fetched %s commits", commits.size());
+        commits = commits.stream()
+                .filter(commit -> !IGNORED_COMMITS_PATTERN.matcher(commit.getTitle()).find())
+                .collect(toImmutableList());
+
+        log.info("Processing %s commits", commits.size());
+        List<PullRequest> pullRequests = commits.stream()
+                .flatMap(commit -> commit.getAssociatedPullRequests().stream())
+                .distinct()
+                .collect(toImmutableList());
+        Map<PullRequest, Optional<List<ReleaseNoteItem>>> releaseNoteItems = pullRequests.stream()
+                .collect(toImmutableMap(identity(), this::extractReleaseNotes));
+
+        log.info("Collecting author information");
+        Map<String, String> userByLogin = new HashMap<>();
+        for (Commit commit : commits) {
+            for (PullRequest pullRequest : commit.getAssociatedPullRequests()) {
+                userByLogin.putIfAbsent(pullRequest.getAuthorLogin(), commit.getAuthor());
+            }
+        }
+        userByLogin = ImmutableMap.copyOf(userByLogin);
+
+        log.info("Generating release notes");
+        String releaseNotes = generateReleaseNotes(releaseNoteItems);
+        String releaseNotesSummary = format(
+                "%s\n%s\n%s",
+                generateMissingReleaseNotes(releaseNoteItems, commits, userByLogin),
+                generateExtractedReleaseNotes(releaseNoteItems, userByLogin),
+                generateCommits(commits));
+
+        log.info("Generating release notes pull request");
+        String releaseNotesBranch = "release-notes-" + version.getVersion();
+        createReleaseNotesCommit(version.getVersion(), releaseNotesBranch, releaseNotes);
+        gitCommands.pushOrigin(releaseNotesBranch);
+
+        PullRequest releaseNotesPullRequest = githubAction.createPullRequest(releaseNotesBranch, format("Add release notes for %s", version.getVersion()), releaseNotesSummary);
+        log.info("Release notes pull request created: %s", releaseNotesPullRequest.getUrl());
+    }
+
+    private Optional<List<ReleaseNoteItem>> extractReleaseNotes(PullRequest pullRequest)
```

Comment:
It looks like some of this PR entry didn't get preserved https://github.com/prestodb/presto/pull/13604 All the lines starting with + got turned into a run-on sentence.

Reply:
Not sure what this comment means?

Reply:
If you run this tool it generates 
![image](https://user-images.githubusercontent.com/1047741/73377331-e13dcd80-428c-11ea-97be-ad22c457f0dc.png)
and
![image](https://user-images.githubusercontent.com/1047741/73377366-f1ee4380-428c-11ea-9990-782cf0672b88.png)
From that PR.

Reply:
This is the expected behavior since we never said we would support sub-bullet-point.